airbyte-webapp/src/components/connection/CatalogTree/next/useRowStatus.tsx

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable css-modules/no-unused-class */
+import { faMinus, faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+import { useField } from "formik";
+import { isEqual } from "lodash";
+import { useMemo } from "react";
+
+import { ModificationIcon } from "components/icons/ModificationIcon";
+import { PillButtonVariant } from "components/ui/PillSelect/PillButton";
+
+import { SyncSchemaStream } from "core/domain/catalog";
+import { useBulkEditSelect } from "hooks/services/BulkEdit/BulkEditService";
+import { useConnectionFormService } from "hooks/services/ConnectionForm/ConnectionFormService";
+
+import styles from "./CatalogTreeTableRow.module.scss";
+
+export const useRowStatus = (stream: SyncSchemaStream) => {
+  const { initialValues } = useConnectionFormService();
+  const [isSelected] = useBulkEditSelect(stream.id);
+
+  const [, { error }] = useField(`schema.streams[${stream.id}].config`);
+
+  // in case error is an empty string
+  const hasError = error !== undefined;
+
+  const isStreamEnabled = stream.config?.selected;
+  const rowStatusChanged =
+    initialValues.syncCatalog.streams.find(
+      (item) => item.stream?.name === stream.stream?.name && item.stream?.namespace === stream.stream?.namespace
+    )?.config?.selected !== stream.config?.selected;
+
+  const rowChanged = !isEqual(
+    initialValues.syncCatalog.streams.find(
+      (item) =>
+        item.stream &&
+        stream.stream &&
+        item.stream.name === stream.stream.name &&
+        item.stream.namespace === stream.stream.namespace
+    )?.config,
+    stream.config
+  );
```

Comment:
This can be moved inside `statusToDisplay` memo since it needs to be computed only for that case